web/src/lib/utils/cookies.ts

```diff
@@ -0,0 +1,30 @@
+import { browser } from '$app/environment';
+
+const IsAuthenticatedCookieName = 'immich_is_authenticated';
+
+export const setAuthCookie = (): void => {
+  if (browser) {
+    document.cookie = `${IsAuthenticatedCookieName}=${String(true)}; Path=/; Max-Age=34560000; SameSite=Lax;`;
+  }
+};
+
+export const getAuthCookie = (): string | null => {
```

Comment:
this should probably just be a private method inside of the file that has `authenticate()` and it can use it to avoid calling getMyInfo when it isn't set.